### PR TITLE
Convert host to canonical name when possible.

### PR DESCRIPTION
**Pull request by jvlcek**
Currently the hostname must be either the canonical hostname or not found in DNS.
This PR will add support allowing the hostname argument to a hostname alias (CNAME)

This is implemented using `Socket.gethostbyname`, which will  return the canonical name
or raise an error if the hostname is not found in DNS.

To test the added functionality confirm a valid configmap is generated for when a host alias (CNAME) is passed to the configmap generate.

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1553790

### CODE REVIEW

**Comment by jvlcek** [comment]
@miq-bot add_label bug

**Comment by jvlcek** [comment]
@miq-bot add_label authentication

**Comment by jvlcek** [comment]
@miq-bot assign @abellotti

**Comment by jvlcek** [comment]
@abellotti Please review. Thank you! JoeV

File: bin/httpd_configmap_generator [hunk 1]
```diff
@@ -62,13 +63,21 @@ supported auth_type: #{HttpdConfigmapGenerator.supported_auth_types.join(', ')}
         end
       end
 
+      params[:host] = get_canonical_hostname(params[:host])
```
Comment (abellotti) [inline bin/httpd_configmap_generator]: I don't think params[:host] is required for an update or export type.  Maybe append that like with if params[:host] ?

File: bin/httpd_configmap_generator [hunk 2]
```diff
@@ -84,4 +93,4 @@ supported auth_type: #{HttpdConfigmapGenerator.supported_auth_types.join(', ')}
   end
 end
 
-HttpdConfigmapGenerator::Cli.new.run
+HttpdConfigmapGenerator::Cli.new.run if $PROGRAM_NAME == __FILE__
```
Comment (abellotti) [inline bin/httpd_configmap_generator]: why was this needed ?

File: bin/httpd_configmap_generator (hunk 1 above)
Comment (jvlcek) [reply bin/httpd_configmap_generator]: 👍 Will do

File: bin/httpd_configmap_generator (hunk 2 above)
Comment (jvlcek) [reply bin/httpd_configmap_generator]: I needed to add that so I could load the file into the spec for testing without running the `#run` method at load time.

**Comment by jvlcek** [comment]
Do not merge. Moving to WIP.  I've learned of a possible situation which might require a different solution. I am researching. Moving this to WIP.

**Comment by jvlcek** [comment]
@miq-bot add_label wip

**Comment by jvlcek** [comment]
Removing this from WIP after more testing.

When Openshift is configured with a routing configuration with a shared canonical
name across multiple MiQ applications/projects then only one IPA configmap is
generated and that same configmap is used on each of the MiQ applications/projects.

@abellotti Please push if you are OK with this.

**Comment by abellotti** [comment]
LGTM!! Thanks @jvlcek for fixing this 👍
